OMEdit/OMEditGUI/Editors/CompositeModelEditor.cpp

```diff
@@ -528,9 +534,15 @@ void CompositeModelEditor::updateConnection(LineAnnotation *pConnectionLineAnnot
     if (connection.attribute("From").compare(pConnectionLineAnnotation->getStartComponentName()) == 0 &&
         connection.attribute("To").compare(pConnectionLineAnnotation->getEndComponentName()) == 0) {
       connection.setAttribute("Delay", pConnectionLineAnnotation->getDelay());
-      connection.setAttribute("alpha", pConnectionLineAnnotation->getAlpha());
-      connection.setAttribute("Zf", pConnectionLineAnnotation->getZf());
-      connection.setAttribute("Zfr", pConnectionLineAnnotation->getZfr());
+      if(!pConnectionLineAnnotation->getAlpha().isEmpty()) {
+        connection.setAttribute("alpha", pConnectionLineAnnotation->getAlpha());
+      }
+      if(!pConnectionLineAnnotation->getZf().isEmpty()) {
+        connection.setAttribute("Zf", pConnectionLineAnnotation->getZf());
+      }
+      if(!pConnectionLineAnnotation->getZfr().isEmpty()) {
+        connection.setAttribute("Zfr", pConnectionLineAnnotation->getZfr());
+      }
```

Comment:
I believe this is wrong.
Assume we have a connection with some `alpha` value and we update the connection and set the `alpha` value to empty. In that case the above code will not update the DOM and we end up having the old `alpha` value in the DOM.